tools/build-tools/src/test/typeValidator/interfaceValidator.spec.ts

```diff
@@ -0,0 +1,391 @@
+            export interface ITestInterface {
+                myProp: number;
+            }
+            `;
+
+            let increment = checkIncrement(sourceOld, sourceNew);
+            assert.strictEqual(increment, BreakingIncrement.minor);
+        });
+    });
+
+    describe("heritage", () => {
+        // extends changes are breaking
+        it("changed interface extensions", () => {
+            const sourceOld =
+            `
+            interface IBaseInterface<T> { myProp: T; }
+            export interface ITestInterface extends IBaseInterface<string> {}
+            `;
+
+            const sourceNew =
+            `
+            interface IBaseInterface<T> { myProp: T; }
+            export interface ITestInterface extends IBaseInterface<number> {}
+            `;
+
+            let increment = checkIncrement(sourceOld, sourceNew);
+            assert.strictEqual(increment, BreakingIncrement.major);
+        });
+
+        // structurally identical extends changes with different names are breaking
+        // (this isn't actually breaking but is difficult to handle and errs on the side
+        // of over-bumping)
+        it("changed interface extensions that are structurally identical", () => {
+            const sourceOld =
+            `
+            interface IBaseInterface<T> { myProp: T; }
+            export interface ITestInterface extends IBaseInterface<string> {}
+            `;
+
+            const sourceNew =
+            `
+            interface IBaseInterface2<T> { myProp: T; }
+            export interface ITestInterface extends IBaseInterface2<string> {}
+            `;
+
+            let increment = checkIncrement(sourceOld, sourceNew);
+            assert.strictEqual(increment, BreakingIncrement.major);
+        });
+
+        // reordering extensions is not breaking
+        it("reorders extensions", () => {
+            const sourceOld =
+            `
+            interface ITestInterface1 {}
+            interface ITestInterface2 {}
+            export class TestClass implements ITestInterface1, ITestInterface2 {}
+            `;
+
+            const sourceNew =
+            `
+            interface ITestInterface1 {}
+            interface ITestInterface2 {}
+            export class TestClass implements ITestInterface2, ITestInterface1 {}
+            `;
+
+            let increment = checkIncrement(sourceOld, sourceNew);
+            assert.strictEqual(increment, BreakingIncrement.none);
+        });
+
+        // Interface extensions of classes are weird in that non-breaking changes
+        // of the extended class can cause downstream compile breaks of others
+        // implementing the interface (e.g. addition of a private method that then
+        // requires implementers of the interface to also inherit from the extended
+        // class).  We don't handle this on the assumption it's incredibly rare
+        it.skip("reflects changes to extended class in implementers", () => {
+        });
```

Comment:
Can we just prevent `interface extends class` altogether in lint or tsconfig?

Reply:
no idea if that already exists, but i guess we could also produce some lint plugin to prevent this and all the other weird long tail features we don't want ._____.